src/veh_interact.cpp

```diff
@@ -873,6 +873,14 @@ bool veh_interact::update_part_requirements()
     }
     nmsg += res.second;
 
+    ret_val<void> can_mount = veh->can_mount(
+                                  veh->part( cpart ).mount, sel_vpart_info->get_id() );
```

Comment:
the main problem comes from here - cpart is only valid some of the time, when on empty square it's probably -1, you can test if veh_interact::dd is the mount point, i can't remember if you need to rotate() it to the correct orientation

looks like adding all parts to req_missing is also misbehaving but that's a smaller problem

Reply:
Yeah looks like it's `dd.rotate( 2 )`. I was so pleased when I saw cpart so I 'could avoid that whole mess'. Shame on me, I suppose.

What's this about adding all parts to req_missing misbehaving?

Reply:
i only checked briefly and not on my pc right now, but some parts had white (installable as opposed to gray) names, but were sorted together with uninstallable ones, if you don't see them in your testing could be just a hiccup, didn't have time to dig into it

Reply:
Oh no, I noticed that right before pushing the PR originally. It's tied to veh_interact::can_potentially_install being used in veh_interact::display_list. ~~I totally ignored that and slammed the PR ahead regardless~~ I gave it some thought over that night and figured it still does provide useful feedback to the player, although it is a slight change from previous behavior. Parts should only be getting colored white if you have all the components to install them, but it's agnostic of whether they're valid to install *on that tile* (would still be able to tell based on their placement in the list).